lib/actions/insert.js

```diff
@@ -1,52 +1,35 @@
+const { messages } = require('elasticio-node');
 const utils = require('../utils.js');
-const pg = require('pg');
+const clientPgPromise = require('../clientPgPromise.js');
 
 /**
  * This method will be called from elastic.io platform providing following data
  *
  * @param msg incoming message object that contains ``body`` with payload
  * @param cfg configuration that is account information and configuration field values
  */
-function processAction(msg, cfg) {
-    const self = this;
-    const conString = cfg.conString;
-    const sql = cfg.query;
-    let script = null; // Cache variable for utils.prepareStatement
-    console.log('Going to merge SQL template with msg sql="%s" body=%j', sql, msg.body);
-    const q = utils.prepareStatement(sql, msg, script);
-    pg.connect(conString, function(err, client, done) {
-        // this initializes a connection pool
-        // it will keep idle connections open for a (configurable) 30 seconds
-        // and set a limit of 20 (also configurable)
-        if (err) {
-            console.error('error fetching client from pool', err);
+async function processAction(msg, cfg) {
+  const self = this;
+  const sql = cfg.query;
+  const script = null; // Cache variable for utils.prepareStatement
+  const q = utils.prepareStatement(sql, msg, script);
 
-            // An error occurred, remove the client from the connection pool.
-            // A truthy value passed to done will remove the connection from the pool
-            // instead of simply returning it to be reused.
-            // In this case, if we have successfully received a client (truthy)
-            // then it will be removed from the pool.
-            done(client);
-            self.emit('error', err);
-            return self.emit('end');
-        }
-        console.log('Executing query q="%s" values=%j', q.text, q.values);
-        const query = client.query(q, function(err, result) {
-            if (err) {
-                console.error('Error when executing the query q="%s" values=%j detail=%s', q.text, q.values, err.detail);
-                done(client);
-                self.emit('error', err);
-                return self.emit('end');
-            }
-            console.log('Query successfully executed, pushing original message further command=%s affectedRows=%s',
-                result.command, result.rowCount);
-            self.emit('data', msg);
-            // End of the processing
-            self.emit('end');
-            // Return connection back to pool
-            done();
-        });
+  const db = clientPgPromise.getDb(cfg);
+  await db.result(q.text, q.values)
```

Comment:
Better to store result in separate variable - change then-catch to await+try-catch
If there's only one 'data' emit in component you can just return message from method

Reply:
> If there's only one 'data' emit in component you can just return message from method

@ElBabbler do you mean, since there is only one 'data' emit in the action, I should be returning *and* emitting it? Or returning the emit? Or just returning the message?

Reply:
Just returning the message